### ISSUE TITLE
add attribution to Grid by ".add_field" method not work as expected. 

### ISSUE THREAD

**Issue by SongshGeo** [JoelForamitti/agentpy#37]
I'm trying to make an inheritance of class "Grid". 

When I try to use the method `add_field()` to add attr to my Grid, I found that from the 3rd row, it works really strange. 
Then, I ran some demo to test: 

```
import agentpy as ap
import numpy as np

model = ap.Model()
shape = (4, 4)
world = ap.Grid(model, shape)
world.add_field("test", 0)
world.test
```

Out[8]: 
array([[     0,      0,      0,      0],
       [     0,      0,      0,      0],
       [999999, 999999, 999999, 999999],
       [999999, 999999, 999999, 999999]])

```
world.add_field("test", np.zeros(shape))
```
world.test2
Out[11]: 
array([[0., 0., 0., 0.],
       [0., 0., 0., 0.],
       [0., 0., 0., 0.],
       [0., 0., 0., 0.]])

```
world.add_field("test3", False)
world.test3
```
Out[13]: 
array([[False, False, False, False],
       [False, False, False, False],
       [ True,  True,  True,  True],
       [ True,  True,  True,  True]])

```
world.add_field("test4", True)
world.test4
```
Out[16]:  
array([[ True,  True,  True,  True],
       [ True,  True,  True,  True],
       [ True,  True,  True,  True],
       [ True,  True,  True,  True]])

**Comment by JoelForamitti** [comment]
Thanks for the report @SongshGeo ! 

A very mysterious bug indeed.

**Comment by JoelForamitti** [comment]
Fixed with 36bf006bc82d81c9c0bfdf2e715451d53cdd323f

**Comment by SongshGeo** [comment]
Thanks for your fantastic project.
